pkg/virt-controller/watch/vm.go

```diff
@@ -619,6 +620,43 @@ func (c *VMController) generateVMIMemoryDumpVolumePatch(vmi *virtv1.VirtualMachi
 	return err
 }
 
+func needUpdatePVCMemoryDumpAnnotation(pvc *k8score.PersistentVolumeClaim, request *virtv1.VirtualMachineMemoryDumpRequest) bool {
+	if pvc.GetAnnotations() == nil {
+		return true
+	}
+	annotation, hasAnnotation := pvc.Annotations[virtv1.PVCMemoryDumpAnnotation]
+	return !hasAnnotation || (request.Phase == virtv1.MemoryDumpUnmounting && annotation != *request.FileName) || (request.Phase == virtv1.MemoryDumpFailed && annotation != memoryDumpFailed)
+}
+
+func (c *VMController) updatePVCMemoryDumpAnnotation(vm *virtv1.VirtualMachine) error {
+	request := vm.Status.MemoryDumpRequest
+	pvc, err := c.getPersistentVolumeClaimFromCache(vm.Namespace, request.ClaimName)
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error getting PersistentVolumeClaim to update memory dump annotation: %v", err)
+		return err
+	} else if pvc == nil {
+		log.Log.Object(vm).Errorf("Error getting PersistentVolumeClaim to update memory dump annotation: %v", err)
+		return fmt.Errorf("Error when trying to update memory dump annotation, pvc %s not found", request.ClaimName)
+	}
+
+	if needUpdatePVCMemoryDumpAnnotation(pvc, request) {
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string, 0))
+		}
+		if request.Phase == virtv1.MemoryDumpUnmounting {
+			pvc.Annotations[virtv1.PVCMemoryDumpAnnotation] = *request.FileName
+		} else if request.Phase == virtv1.MemoryDumpFailed {
+			pvc.Annotations[virtv1.PVCMemoryDumpAnnotation] = memoryDumpFailed
+		}
+		pvc, err = c.clientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(context.Background(), pvc, v1.UpdateOptions{})
```

Comment:
I'd remove the `pvc` variable as you aren't using it anymore and evaluate the if on a single line
```golang
if _, err = c.clientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(context.Background(), pvc, v1.UpdateOptions{})`; err != nil
```